Library/Homebrew/dev-cmd/tests.rb

```diff
@@ -49,6 +49,8 @@ def tests
       ENV["HOMEBREW_TEST_GENERIC_OS"] = "1" if args.generic?
       ENV["HOMEBREW_TEST_ONLINE"] = "1" if args.online?
 
+      ENV["USER"] ||= Utils.popen_read("id", "-nu").chomp
```

Comment:
Use `system_command!("id", args: ["-nu"]).stdout.chomp` instead.

Reply:
`system_command` doesn't observe `HOMEBREW_STDERR`, so I prefer to use `Utils.popen_read`.
https://github.com/Homebrew/brew/blob/4d68bee1be06625251afcf340030caffef1ec4cd/Library/Homebrew/utils/popen.rb#L31
What is the advantage of `system_command` over `Utils.popen_read`?

Reply:
> What is the advantage of `system_command` over `Utils.popen_read`?

`system_command!` raises an exception when the command fails, which also includes the content of `stderr`.

The goal would be to replace all the `popen*` commands with `system_command`.

Reply:
> `system_command!` raises an exception when the command fails, which also includes the content of `stderr`.

Ah, that's a nice feature!

`system_command!` is more heavyweight (using `popen3` and `select` to poll multiple file descriptors) than `Utils.popen_read`. Perhaps something to keep in mind for larger I/O jobs.

Reply:
> The goal would be to replace all the `popen*` commands with `system_command`.

@reitermarkus Would be worth opening a `help wanted` issue for that?